webapp/src/Entity/Language.php

```diff
@@ -189,6 +193,38 @@ public function getCompileExecutableHash(): ?string
         return $this->compile_executable?->getImmutableExecutable()->getHash();
     }
 
+    #[Serializer\VirtualProperty]
+    #[Serializer\SerializedName('compiler')]
+    #[Serializer\Exclude(if:'object.getCompilerVersionCommand() == ""')]
+    public function getCompilerData(): ?array
+    {
+        $ret = [];
+        if (!empty($this->getCompilerVersionCommand())) {
+            $ret['version_command'] = $this->getCompilerVersionCommand();
+            if (!empty($this->getCompilerVersion())) {
+                $ret['version'] = $this->getCompilerVersion();
+            }
+        } else {
+            return null;
+        }
+        return $ret;
+    }
```

Comment:
```suggestion
        $ret = [];
        if (!empty($this->getCompilerVersionCommand())) {
            $ret['version_command'] = $this->getCompilerVersionCommand();
            if (!empty($this->getCompilerVersion())) {
                $ret['version'] = $this->getCompilerVersion();
            }
            return $ret;
        }
        return null;
    }
```

Or do we expect another case in the future?

Reply:
No `null` anymore.